### PR TITLE
Create new tank armory

### PR DESCRIPTION
**Pull request by JorgeVidal29**
This creates a new tank for the user from the armory

**Description**
Creates a new tank for the user functionality added to armory

Resolves #(replace parentheses and the stuff here with the issue number, if there isn't one, create one.) 

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
Make a bulleted/numbered list of steps to verify this feature/bugfix. Include screenshots if necessary.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: src/armory/Armory.js [hunk 1]
```diff
@@ -113,7 +115,6 @@ class Armory extends React.Component<Props, State> {
 			},
 		});
 		const body = await response.text();
-		console.log(body);
```
Comment (SecondThread) [inline src/armory/Armory.js:116]: thanks :)

File: src/armory/Armory.js [hunk 2]
```diff
@@ -113,7 +115,6 @@ class Armory extends React.Component<Props, State> {
 			},
 		});
 		const body = await response.text();
```
Comment (SecondThread) [inline src/armory/Armory.js]: change to response.json() please

File: src/armory/Armory.js [hunk 3]
```diff
@@ -286,19 +287,19 @@ class Armory extends React.Component<Props, State> {
 		return (
 			<div id="Parent">
 				<Navbar styleName="navbtn" linkName="MainMenu" returnName="Back to Main Menu" pageName="Armory" userName="FRIcker | $465128" />
-				<div className="column armoryleft">
-					<h3>Select a Tank to Edit</h3>
-					<select className="dropdownMenu" value={this.state.selectedTankId} onChange={this.handleChangeInTankOptions}>{tankOptions.map(({ value, label }, index) => <option key={index}  value={value}>{label}</option>)}</select>
 					<div className="column armoryleft">
+						<h3>Select a Tank to Edit</h3>
+						<select className="dropdownMenu" value={this.state.selectedTankId} onChange={this.handleChangeInTankOptions}>{tankOptions.map(({ value, label }, index) => <option key={index}  value={value}>{label}</option>)}</select>
```
Comment (SecondThread) [inline src/armory/Armory.js]: This is a really long line

File: src/armory/CreateNewTankPopup.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+|};
+
+class CreateNewTankPopup extends React.Component<Props, State> {
+
+	constructor() {
+		super();
+		this.state={
+            newTankName: '',
+            errorMessage: '',
+            newTankDialogOpen: false,
```
Comment (SecondThread) [inline src/armory/CreateNewTankPopup.js]: fix whitespace please

**Review by SecondThread — CHANGES_REQUESTED** [review]
Doesn't pass flow. Also the cancel button doesn't close the dialog. Also if you click outside the dialog you can't open it a second time. (All three of those are caused by the same thing, but I won't spoil that because I think it is a good learning experience and shows why flow is so helpful)

File: src/armory/Armory.js (hunk 3 above)
Comment (JorgeVidal29) [reply src/armory/Armory.js]: https://stackblitz.com/edit/react-select-example this is where I got that idea

File: src/armory/Armory.js [hunk 5]
```diff
@@ -281,48 +280,119 @@ class Armory extends React.Component<Props, State> {
 		});
 	};
 
-
+	//The || in the values are for a warning read more here https://github.com/mui-org/material-ui/issues/4904
```
Comment (SecondThread) [inline src/armory/Armory.js]: If you read the issue linked it says that this issue isn't actually a problem with Material Design. But also you shouldn't be passing null in there anyway...

File: src/armory/CreateNewTankPopup.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+		super();
+		this.state={
+ 			newTankName: '',
+			errorMessage: '',
+			newTankDialogOpen: false,
+		}
+    }
+	
+	//Creates a new tank
+    handleCreateClick(): void {
```
Comment (SecondThread) [inline src/armory/CreateNewTankPopup.js]: please fix spacing

File: src/armory/CreateNewTankPopup.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+							{cancelButton}
+						</div>
+					</div>
+				</Popup>
+			</div>
+		);
+	}
+}
+
+export default CreateNewTankPopup;
```
Comment (SecondThread) [inline src/armory/CreateNewTankPopup.js:113]: Thanks, this file looks good

File: src/armory/Armory.js (hunk 5 above)
Comment (SecondThread) [reply src/armory/Armory.js]: We talked about it offline, but this is hiding a warning from us that we want to see if it happening. I can't reproduce it easily on my end, but we want to see these warnings when someone refactors this

File: src/armory/Armory.js [hunk 8]
```diff
@@ -281,48 +280,119 @@ class Armory extends React.Component<Props, State> {
...
-					<h3>Select a Tank to Edit</h3>
-					<select className="dropdownMenu" value={this.state.selectedTankId} onChange={this.handleChangeInTankOptions}>{tankOptions.map(({ value, label }, index) => <option key={index}  value={value}>{label}</option>)}</select>
 					<div className="column armoryleft">
+						<h3>Select a Tank to Edit</h3>
+						<select 
+							className="dropdownMenu" 
+							value={this.state.selectedTankId} 
+							onChange={this.handleChangeInTankOptions}
+						>
+							{tankOptions.map(({ value, label }, index) => <option key={index}  value={value}>{label}</option>)}
```
Comment (SecondThread) [inline src/armory/Armory.js:294]: I know this isn't part of the changes you made here (so you don't have to fix it before pushing, just something to keep in mind): the way that this stuff is stored with the value/label/index is pretty confusing

File: src/armory/Armory.js [hunk 9]
```diff
@@ -281,48 +280,119 @@ class Armory extends React.Component<Props, State> {
...
-						<h1>BIG TANK GUY</h1>
+						<h1>{this.state.selectedTankName}</h1>
 						<h6>Points Used: 0/10</h6>
 					</div>
 					<div className="column armoryright">
 						<h6>Chassis</h6>
-						<select className="tankComponentMenu" value={this.state.selectedChassis} onChange={this.handleChangeInChassisOptions}>{chassisOptions.map(({ value, label  }, index) => <option key={index} value={value}>{label}</option>)}</select>
+						<select 
+							className="tankComponentMenu"
+							value={this.state.selectedChassis || ''}
```
Comment (SecondThread) [inline src/armory/Armory.js]: Talked about offline, but make sure you delete this `|| ''` and all similar ones before pushing

**Review by SecondThread — APPROVED** [review]
Make sure to fix PR comments before pushing

File: src/armory/Armory.js (hunk 5 above)
Comment (JorgeVidal29) [reply src/armory/Armory.js]: Yes to future refactors, there is an issue when you select a tank that doesn't have a selected component. index.js:1 Warning: `value` prop on `select` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.

File: src/armory/CreateNewTankPopup.js (hunk 7 above)
Comment (JorgeVidal29) [reply src/armory/CreateNewTankPopup.js:113]: Thank you

**Pull request merged**